### ISSUE TITLE
corss_entropy result diffrence from python 

### ISSUE THREAD

**Issue by amdcpuwu** [rstudio/tensorflow#222]
i use the same code in R use tensorflow,but the resulte very diffrence from python 

below is my code
`

> library(tensorflow)
> batch_size<-8
> w1<-tf$Variable(tf$random_normal(list(2L,3L),stddev = 1,seed = 1))
> w2<-tf$Variable(tf$random_normal(list(3L,1L),stddev = 1,seed = 1))
> x<-tf$placeholder(tf$float32,shape=list(NULL,2L),name = 'x-input')
> y_<-tf$placeholder(tf$float32,shape=list(NULL,1L),name='y-input')
> a<- tf$matmul(x,w1) 
> y<-tf$matmul(a,w2)
> cross_entropy<- -tf$reduce_mean(y_ * tf$log(tf$clip_by_value(y,1e-10,1.0)))
> train_step<-tf$train$AdamOptimizer(0.001)$minimize(cross_entropy)
> set.seed(1)
> X<-array_reshape(rnorm(256,0,1),c(128,2))
> x1<-X[,1]
> x2<-X[,2]
> Y <-array_reshape(as.integer(x1+x2<1),c(128,1))
> sess<-tf$Session()
> sess$run(tf$global_variables_initializer())

> for (i in 1:7000) {
  start<- ( i + 8L) %% 128L
  end<- min(start+8L,128L)
 sess$run(train_step,feed_dict = dict(x=X[start:end,],y_= matrix(Y[start:end],length(Y[start:end]),1)))
  if (i %% 1000 ==0){
    Total_cross_entropy<-sess$run(cross_entropy,feed_dict = dict(x=X,y_=Y))
    cat(sprintf("step %d, Total_cross_entropy %g\n", i , Total_cross_entropy))
  }
}`
output:
step 1000, Total_cross_entropy 11.4266
step 2000, Total_cross_entropy 11.5833
step 3000, Total_cross_entropy 11.5875
step 4000, Total_cross_entropy 11.4274
step 5000, Total_cross_entropy 11.4163
step 6000, Total_cross_entropy 11.4137
step 7000, Total_cross_entropy 11.4209

Python:
``

> import tensorflow as tf
> from numpy.random import RandomState
> 
> batch_size = 8
> w1=tf.Variable(tf.random_normal([2,3],stddev=1,seed=1))
> w2=tf.Variable(tf.random_normal([3,1],stddev=1,seed=1))
> x=tf.placeholder(tf.float32,shape=(None,2),name='x-input')
> y_=tf.placeholder(tf.float32,shape=(None,1),name='y-input')
> a=tf.matmul(x,w1)
> y=tf.matmul(a,w2)
> cross_entropy=-tf.reduce_mean(y_ * tf.log(tf.clip_by_value(y,1e-10,1.0)))
> train_step=tf.train.AdamOptimizer(0.001).minimize(cross_entropy)
> rdm = RandomState(1)
> dataset_size = 128

> X = rdm.rand(dataset_size, 2)
> Y=[[int(x1+x2<1)] for (x1,x2) in X]
> sess=tf.Session()
> init_op=tf.global_variables_initializer()
> sess.run(init_op)
> print(sess.run(w1))
> print(sess.run(w2))
> STEPS = 7000

> for i in range(STEPS):
        start = (i * batch_size) % dataset_size
        end=min(start+batch_size,dataset_size)
        sess.run(train_step,feed_dict = { x:X[start:end],y_:Y[start:end]})
        if i % 1000 == 0:
            total_cross_entropy = sess.run(cross_entropy,feed_dict={x:X,y_:Y})
            print("step %d, total_cross_entropy %g" % (i, total_cross_entropy))`
``

output:
step 0, total_cross_entropy 0.0674925
step 1000, total_cross_entropy 0.0163385
step 2000, total_cross_entropy 0.00907547
step 3000, total_cross_entropy 0.00714436
step 4000, total_cross_entropy 0.00578471
step 5000, total_cross_entropy 0.00430222
step 6000, total_cross_entropy 0.00280812
